### PR TITLE
Update cpprestsdk to 2.10.16

### PR DESCRIPTION
**Pull request by garethsb-sony**
Specify library name and version:  **cpprestsdk/2.10.16**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/cpprestsdk/all/conandata.yml [hunk 1]
```diff
@@ -1,9 +1,9 @@
 sources:
-  "2.10.15":
-    sha256: 1c027a53457e87b0b3a475e5c8045b94400c475898c8bd51b0fbd218b99a7f7b
-    url: https://github.com/Microsoft/cpprestsdk/archive/v2.10.15.tar.gz
+  "2.10.16":
```
Comment (SSE4) [inline recipes/cpprestsdk/all/conandata.yml:2]: please keep old version in place

Comment (garethsb-sony) [reply recipes/cpprestsdk/all/conandata.yml:2]: OK... This will need separate versions of the patches per version.

**Comment by prince-chrismc** [comment]
Totally missed this #1459 did the same work myself

**Comment by garethsb-sony** [comment]
@prince-chrismc, no probs, I'll close in favour of #1459. I was going to be put a sleep/retry loop round the `os.rename` that causes issues on several of my Windows environments, but never mind!

**Comment by prince-chrismc** [comment]
I'm shocked to hear that, was the download the problem? Could you share the error?

**Comment by garethsb-sony** [comment]
Something on Windows often prevents rename of the source folder. However I see I'm obviously not the first to encounter this (https://github.com/conan-io/conan/pull/5639 and several issues in the trackers). In my case, indexer service and most other suggested culprits disabled. I'll just keep patching the recipes locally.

**Comment by prince-chrismc** [comment]
Ahhh, I am familiar with the anti virus being a culprit.

**Pull request closed without merging**